### DISCUSSION TITLE
Is it possible for one property in a model to re-use the value of another?

### DISCUSSION THREAD

**Discussion by wrex** [mswjs/data#157] [Q&A] [answered]
I'm writing some tests and would like to model a REST API out of my control. GET requests to that API return JSON objects like:

```json
{
  "id": 534342,
  "object": "review",
  "url": "https://api.wanikani.com/v2/reviews/534342",
  "data_updated_at": "2017-12-20T01:00:59.255427Z",
  "data": {
    "created_at": "2017-12-20T01:00:59.255427Z",
    ...
  }
}
```

The URL value always includes the ID value, which is redundant and not normalized but I'd like to model it nonetheless.

Is there a reliable and idiomatic way for a factory to return two properties with the same value within both? I'd like to use the moral equivalent of `this.id` in the parameters to the factory, but I'm unsure how to go about it. Something like the following:

```js
const db = factory({
  user: {
    id: primaryKey(() => datatype.number({max: 999999})),
    url: () => `https://api.wanikani.com/v2/reviews/{this.id}`,  // doesn't work, of course
    ...
  },
})
```

Thanks in advance!

**Comment by kettanaito** [comment, marked as answer]
Hey, @wrex. Thank you for raising this!

Currently, we don't have an API to allow referencing other properties of the same entity in the value getters. There's, however, a suggestion in https://github.com/mswjs/data/issues/95. We would like to support this, so please share with us any ideas/suggestions on how you want this API to work in the said issue. 

Ideally, I'd like to see an API like this:

```js
factory({
  user: {
    id: primaryKey(...),
    url: (entity) => entity.id
  }
})
```

**Reply by wrex** [reply]
Yes, passing the current entity as a parameter would work great for my simple use-case. I'll redirect further comments to issue #95.

Also: Thanks for this useful library! It's proving very useful to me even outside of an msw context (where I'm mocking a higher level client-side library that takes care of caching and pagination with the server-side API).

**Reply by kettanaito** [reply]
I would be grateful for your input! 

Thank you for your kind words. With Data we strive to design a standalone tool in the first place, ensure smooth integration with MSW as a nice feature later. Please, if you would like for this library to improve, consider supporting our team at [Open Collective](https://opencollective.com/mswjs). Any contribution counts.
